### PR TITLE
create default configurations for defender library

### PR DESCRIPTION
**Pull request by qiutongs**
Description
-----------

Create necessary compile-time required/default configurations for defender library.

Checklist:
----------

- [NO] I have tested my changes. No regression in existing tests.
- [X] My code is Linted.

### CODE REVIEW

File: lib/include/private/aws_iot_defender_config_defaults.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef _AWS_IOT_DEFENDER_CONFIG_DEFAULT_H_
+#define _AWS_IOT_DEFENDER_CONFIG_DEFAULT_H_
+
+#include "aws_clientcredential.h"
```
Comment (gordonwang0) [inline lib/include/private/aws_iot_defender_config_defaults.h]: These files won't be available on the C SDK -- please keep that in mind

**Comment by gordonwang0** [comment]
For the v4 libraries, default configurations go in an internal library header, i.e. `aws_iot_defender_internal.h`.  This is fine for now, but you'll have to change it later.

File: lib/include/private/aws_iot_defender_config_defaults.h (hunk 1 above)
Comment (qiutongs) [reply lib/include/private/aws_iot_defender_config_defaults.h]: I see

**Comment by qiutongs** [comment]
> For the v4 libraries, default configurations go in an internal library header, i.e. `aws_iot_defender_internal.h`. This is fine for now, but you'll have to change it later.

Thanks for letting me know. I will change it soon.

**Pull request merged**